### PR TITLE
Implement getListingByID

### PR DESCRIPTION
**Pull request by HellinSun**
This pull request brings the following:

Additions:

- getLisingByID is implemented so that a certain product could be fetched from backend given its product ID.

- The product page has been updated to show the all detail information(including all the product info, product images, contributor info etc.) of the product clicked from AR, Map and Feed pages.

Changes:

- Rename ToDataURI to DataURIHelper for consistency.
- Rename SliderAdapter to ProductPicsSliderAdapter.
- Changed condition and category options according to backend settings
- Add condition attribute to Product class

### CODE REVIEW

**Review by Manoutofpower — APPROVED** [review]

File: app/src/main/java/com/example/ar_reshare/AddProduct.java [hunk 1]
```diff
@@ -168,7 +158,7 @@ public void onClick(View v) {
                         ArrayList<String> dataURIList;
                         dataURIList = convertToDataURI(adapter.uploadedImages);
                         media.addAll(dataURIList);
-                        BackendController.addProduct(productName,productDescription,"UK","Bristol",productPostcode, 1,"new", media, AddProduct.this);
+                        BackendController.addProduct(productName,productDescription,"UK","Bristol",productPostcode,category,condition, media, AddProduct.this);
```
Comment (artur-varosyan) [inline app/src/main/java/com/example/ar_reshare/AddProduct.java:161]: Hey, this is just a suggestion but when the user types in the postcode, if you use the _PostcodeHelper_ you can find the region the postcode belongs to (_so for example Bristol postcodes are "Bristol, City of"_). So maybe in the future, we could replace the hardcoded _"Bristol"_ value with a dynamically determined region? Let me know what you think!

**Review by artur-varosyan — APPROVED** [review]
Great job Hellin. Lots of new features and I love how clean and structured the code is!

**Review by Lingtao-Cai — APPROVED** [review]
really clean and neat structure, nice job!

**Review by arfi2212 — APPROVED** [review]
This looks great, thank you Hellin! :)

**Pull request merged**